### ISSUE TITLE
Upgrade to wxWidgets 3.1?

### ISSUE THREAD

**Issue by alexey-lysiuk** [sirjuddington/SLADE#469]
wxWidgets 3.1 was released a few days ago. Is an upgrade to this version planned?

I tried to build with it on OS X but it required some minor changes in order to compile. Also, the recent items on the start page appeared to be broken after the upgrade :disappointed:

At the same time there are lots of improvements according to wxWidgets release notes. For instance, I noticed that HiDPI screens are now supported in the text editor.

**Comment by sirjuddington** [comment]
Yeah, I've updated the VS solution to use 3.1 in master, and considering it for stable. The improvements are pretty nice (although the high DPI stuff is kinda buggy in Windows 10).

I'm going to keep 3.0 as the minimum required to build, though, since 3.1 is labelled as a 'development' release like 2.9 was, which means it won't likely end up on any linux package repositories until 3.2 (and who knows how many years that will take).

So basically 3.1 is fine, just don't use any 3.1-specific features in the code.
